flakehub_push crashes if rolling-prefix is not set and a tag is not available

With no `rolling-prefix` set and not tag pushed flakehub_push crashes when the action runs. 

Here's the example action:

```yaml
name: Push to FlakeHub

on:
  push:
    branches: [main]
  workflow_dispatch:

jobs:
  push:
    runs-on: ubuntu-22.04
    permissions:
      id-token: write
      contents: read
    steps:
      - uses: actions/checkout@v3
      - uses: DeterminateSystems/nix-installer-action@v4
      - uses: DeterminateSystems/flakehub-push@main
        with:
          upload-name: DeterminateSystems/obs-cli-linux
          visibility: hidden
```

Here is the log:

```
Set FLAKEHUB_PUSH_URL=https://install.determinate.systems/flakehub-push/branch/main/X64-Linux?ci=github
  % Total    % Received % Xferd  Average Speed   Time    Time     Time  Current
                                 Dload  Upload   Total   Spent    Left  Speed

  0     0    0     0    0     0      0      0 --:--:-- --:--:-- --:--:--     0
  0     0    0     0    0     0      0      0 --:--:-- --:--:-- --:--:--     0

  0 6241k    0     0    0     0      0      0 --:--:-- --:--:-- --:--:--     0
100 6241k  100 6241k    0     0  11.5M      0 --:--:-- --:--:-- --:--:-- 46.1M
2023-08-08T14:15:57.912265Z  INFO flakehub_push:push_new_release{repository=DeterminateSystems/obs-cli-linux mirror=false upload_name=DeterminateSystems/obs-cli-linux}: flakehub_push::cli: Preparing release of DeterminateSystems/obs-cli-linux/main
Error: 
   0: Status 400 Bad Request from metadata POST
      Invalid URL: unexpected character 'm' while parsing major version number

Location:
   src/cli/mod.rs:436

  ━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━ SPANTRACE ━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━

   0: flakehub_push::cli::push_new_release with repository=DeterminateSystems/obs-cli-linux mirror=false upload_name=DeterminateSystems/obs-cli-linux source=/nix/store/imqvsv6d2mc9fabx2r7kjbpm4njmaq5k-source
      at src/cli/mod.rs:267
   1: flakehub_push::cli::flakehub_push
      at src/cli/mod.rs:120

Backtrace omitted. Run with RUST_BACKTRACE=1 environment variable to display it.
Run with RUST_BACKTRACE=full to include source snippets.

Consider reporting this error using this URL: /issues/new
Error: Process completed with exit code 1.
```

- Action Log: https://github.com/DeterminateSystems/obs-cli-linux/actions/runs/5798109518/job/15715138432

## Expectation

It would be better if this condition is caught and useful information is output to guide the user to the appropriate corrective action.

--- a/Cargo.toml
+++ b/Cargo.toml
@@ -29,6 +29,7 @@ atty = { version = "0.2.14", default-features = false }
 github-actions-oidc-claims = "0.3.0"
 spdx = "0.10.2"
 uuid = { version = "1.4.0", features = ["serde", "v7", "rand", "std"] }
+semver = { version = "1.0.18", features = ["serde"] }
 
 [profile.release]
 strip = true  # Automatically strip symbols from the binary.

--- a/src/cli/mod.rs
+++ b/src/cli/mod.rs
@@ -4,7 +4,8 @@ use color_eyre::eyre::{eyre, WrapErr};
 use reqwest::{header::HeaderMap, StatusCode};
 use std::{
     path::{Path, PathBuf},
-    process::ExitCode, str::FromStr,
+    process::ExitCode,
+    str::FromStr,
 };
 use tokio::io::AsyncWriteExt;
 use uuid::Uuid;
@@ -469,7 +470,7 @@ async fn push_new_release(
             // Ensure the version respects semver
             semver::Version::from_str(&tag).wrap_err_with(|| eyre!("Failed to parse version `{tag}` as semver, see https://semver.org/ for specifications"))?;
             tag
-        },
+        }
         (None, None) => {
             return Err(eyre!("Could not determine tag or rolling minor version, `--tag`, `GITHUB_REF_NAME`, or `--rolling-minor` must be set"));
         }

--- a/src/cli/mod.rs
+++ b/src/cli/mod.rs
@@ -4,7 +4,7 @@ use color_eyre::eyre::{eyre, WrapErr};
 use reqwest::{header::HeaderMap, StatusCode};
 use std::{
     path::{Path, PathBuf},
-    process::ExitCode,
+    process::ExitCode, str::FromStr,
 };
 use tokio::io::AsyncWriteExt;
 use uuid::Uuid;
@@ -465,7 +465,11 @@ async fn push_new_release(
         }
         (Some(minor), _) => format!("0.{minor}"),
         (None, _) if rolling => DEFAULT_ROLLING_PREFIX.to_string(),
-        (None, Some(tag)) => tag,
+        (None, Some(tag)) => {
+            // Ensure the version respects semver
+            semver::Version::from_str(&tag).wrap_err_with(|| eyre!("Failed to parse version `{tag}` as semver, see https://semver.org/ for specifications"))?;
+            tag
+        },
         (None, None) => {
             return Err(eyre!("Could not determine tag or rolling minor version, `--tag`, `GITHUB_REF_NAME`, or `--rolling-minor` must be set"));
         }

